WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -262,6 +262,64 @@ def edit
     render :edit
   end
 
+  def payment_setup
+    @competition = Competition.find(params[:id])
+
+    client = create_oauth_client
+    oauth_params = {
+      scope: 'read_write',
+      state: @competition.id,
+    }
+    @authorise_url = client.auth_code.authorize_url(oauth_params)
+  end
```

Comment:
It's probably worth noting that this rewrite may have an injection vulnerability thanks to using `params[:id]`. I feel much more comfortable doing a database query to find the competition first.